aksel.nav.no/website/components/website-modules/utils/amplitude.tsx

```diff
@@ -72,46 +80,95 @@ export const usePageView = (router: Router, pageProps: any) => {
         isDevelopment && console.error(error);
       }
       logPageView(e, data, first);
+      try {
+        if (isForside && isProduction()) {
+          fetch(`/api/log-page-view?id=${pageId}`);
+        }
+      } catch (error) {
+        isDevelopment && console.error(error);
+      }
     },
-    [pageProps]
+    [pageProps, pageId, isForside]
   );
 
-  /* https://stackoverflow.com/questions/2387136/cross-browser-method-to-determine-vertical-scroll-percentage-in-javascript */
-  const logScroll = useCallback(() => {
-    function getScrollPercent() {
-      const h = document.documentElement,
-        b = document.body,
-        st = "scrollTop",
-        sh = "scrollHeight";
-      return ((h[st] || b[st]) / ((h[sh] || b[sh]) - h.clientHeight)) * 100;
-    }
-    if (
-      document === undefined ||
-      window?.location?.pathname?.startsWith?.("/eksempler")
-    ) {
-      return;
-    }
+  const logScroll = useCallback(
+    (highestPercent: number) => {
+      if (
+        document === undefined ||
+        window?.location?.pathname?.startsWith?.("/eksempler")
+      ) {
+        return;
+      }
 
-    const scrollD = Math.round(getScrollPercent() / 10) * 10;
-    if (isNaN(scrollD)) {
-      return;
-    }
+      if (isNaN(highestPercent)) {
+        return;
+      }
 
-    logAmplitudeEvent(AmplitudeEvents.scroll, {
-      side: window.location.pathname,
-      prosent: scrollD,
-    });
-  }, []);
+      logAmplitudeEvent(AmplitudeEvents.scroll, {
+        side: window.location.pathname,
+        prosent: highestPercent,
+      });
+
+      try {
+        if (isForside && isProduction()) {
+          fetch(`/api/log-scroll?id=${pageId}&length=${highestPercent}`);
+        }
+      } catch (error) {
+        isDevelopment && console.error(error);
+      }
+    },
+    [pageId, isForside]
+  );
+
+  const logTimeSpent = useCallback(
+    (timeSpent: number) => {
+      try {
+        if (isForside && timeSpent <= 420 && isProduction()) {
+          fetch(`/api/log-time?id=${pageId}&time=${timeSpent}`);
+        }
+      } catch (error) {
+        isDevelopment && console.error(error);
+      }
+    },
+    [pageId, isForside]
+  );
 
   useEffect(() => {
+    const startTime = new Date().getTime();
+
+    let highestPercent = 0;
+    let timeoutId = null;
+
+    //get highest scroll percent
+    function scrollListener() {
```

Comment:
Ser ut som en god løsning for scroll 💯 Husker det var nevnt å kanskje ignorere footer-høyden, men vet ikke om vi ønsket å gjøre det?

Reply:
Kan godt få inn det!